addon/-private/utils/observer.js

```diff
@@ -0,0 +1,50 @@
+import { gte } from 'ember-compatibility-helpers';
+import { assert } from '@ember/debug';
+
+// eslint-disable-next-line no-restricted-imports
+import { observer as eObserver } from '@ember/object';
+
+// eslint-disable-next-line no-restricted-imports
+import {
+  addObserver as eAddObserver,
+  removeObserver as eRemoveObserver,
+} from '@ember/object/observers';
+
+const USE_ASYNC_OBSERVERS = gte('3.13.0');
+
+function asyncObserver(...args) {
+  let fn = args.pop();
+  let dependentKeys = args;
+  let sync = false;
+
+  return eObserver({ dependentKeys, fn, sync });
```

Comment:
This appears to be an acceptable way to call `observer`, see

https://github.com/emberjs/ember.js/blob/133609c7766ef9c02060a07d8fd885f7d2f1940a/packages/%40ember/-internals/metal/lib/mixin.ts#L868
and 
https://github.com/emberjs/ember.js/blob/133609c7766ef9c02060a07d8fd885f7d2f1940a/packages/%40ember/-internals/metal/lib/mixin.ts#L840

It is not part of the docs yet, but I think that is simply because they haven't been updated.